src/transformers/modeling_utils.py

```diff
@@ -3224,6 +3231,15 @@ def from_pretrained(
         if quantization_method_from_config == QuantizationMethod.GPTQ:
             model = quantizer.convert_model(model)
             model._is_quantized_training_enabled = True
+        elif quantization_method_from_config == QuantizationMethod.AWQ:
+            from .integrations import replace_with_awq_linear
+
+            if quantization_config is None:
+                quantization_config = AWQConfig.from_dict(config.quantization_config)
+
+            model, _ = replace_with_awq_linear(
+                model, quantization_config=quantization_config, modules_to_not_convert=["lm_head"]
```

Comment:
The `modules_to_not_convert=["lm_head"]` may end up causing issues eventually if the head is not named "lm_head" exactly. The way we deal with avoiding this is by only looking at the decoder layers of the model in AutoAWQ by calling the model's `get_model_layers()` function (e.g. [llama](https://github.com/casper-hansen/AutoAWQ/blob/main/awq/models/llama.py#L17))

Reply:
Nice! For bnb we have the same issue and use this method: https://github.com/huggingface/transformers/blob/main/src/transformers/integrations/bitsandbytes.py#L243 which should be quite generic for most `transformers` models. I planned to use that instead